src/func.cc

```diff
@@ -576,8 +584,10 @@ void ShellFunc(const vector<Value*>& args, Evaluator* ev, string* s) {
   const string&& shellflag = ev->GetShellFlag();
 
   string out;
+  // The 77 should be overridden in ShellFuncImpl
```

Comment:
...then why set a value here?

Reply:
Au contraire. Crashing immediately is much better than silently using the wrong error code in case this "77" is not overwritten.

If you crash immediately, it's obvious where the problem is and that "77" can propagate who-knows-where and then one would have to painstakingly trace the chain of events back here.

Reply:
I changed ShellFuncImpl to return the returnCode instead of passing it out via an argument, so we don't hit this issue anymore.